### PR TITLE
APIGOV-27136 - add guidance on how invoke policy auth is integrated with agent

### PR DESCRIPTION
**Pull request by jcollins-axway**
## Describe the changes

- Add guidance on how invoke policy auth is integrated with agent. 
- Additionally add a note on IDP configuration with the use of Client Registration Policies on the IDP.

## Deploy preview link

- http://pr-397.amplify-central.opendocs-builder.pcloud.axway.int/docs/connect_manage_environ/connect_api_manager/gateway-administation/#invoke-policy-authentication-handling-in-the-discovery-agent
- http://pr-397.amplify-central.opendocs-builder.pcloud.axway.int/docs/connect_manage_environ/connect_api_manager/agent-variables/#common-variables-to-both-agents
- http://pr-397.amplify-central.opendocs-builder.pcloud.axway.int/docs/connect_manage_environ/marketplace_provisioning/#provisioning-oauth-credential-to-an-identity-provider

## Checklist for contributors

Before submitting this PR:

* Verify that all status checks have passed
* For more information on the Markdown linter rules, see [DavidAnson/markdownlint](https://github.com/DavidAnson/markdownlint)
* For first-time contributors, ensure that you have signed the [Axway CLA](https://cla.axway.com/)

### CODE REVIEW

**Comment by docsbuilder** [comment]
### <span aria-hidden="true">✅</span> Deploy preview is ready!

 |  Name | Link |
 |---------------------------------|------------------------|
 |<span aria-hidden="true">🔨</span> Latest commit | 86853be |
 |<span aria-hidden="true">😎</span> Deploy preview | http://PR-397.amplify-central.opendocs-builder.pcloud.axway.int |

File: content/en/docs/connect_manage_environ/connect_api_manager/agent-variables.md [hunk 1]
```diff
@@ -98,6 +98,7 @@ You can extend the previous minimum variable list with the following variables.
...
 | APIMANAGER_WORKERS                                                 | Controls how many Front End Proxies will be processed at a single time (default: 10).                                                                                                                                                                                                                                                                                                                                                                                                                      |
+| APIMANAGER_INVOKEPOLICYDEFAULTDESCRIPTION                          | When a proxy is secured by a policy this description is added to the Access Request Definition if no desription text is found in API Manager (default: `Contact your provider about authenticating to this proxy`)                                                                                                                                                                                                                                                                                         |
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connect_api_manager/agent-variables.md:101]: Comma after between policy and this

Comment (jcollins-axway) [reply content/en/docs/connect_manage_environ/connect_api_manager/agent-variables.md:101]: fixed

File: content/en/docs/connect_manage_environ/connect_api_manager/gateway-administation.md [hunk 2]
```diff
@@ -195,6 +195,14 @@ When using OAuth (External) inbound security, API Manager requires a custom OAut
 
 When discovering and publishing the virtualized APIs with OAuth (External) inbound security, the Discovery Agent can associate the registered external OAuth identity providers to the published resources on Enterprise Marketplace that allows Marketplace consumers to provision credential to specified OAuth identity provider. For details on how to register external OAuth identity providers, see [Provisioning OAuth credential to an identity provider](/docs/connect_manage_environ/marketplace_provisioning/#provisioning-oauth-credential-to-an-identity-provider)
 
+#### Invoke Policy authentication handling in the Discovery Agent
+
+When virtualizing a REST API in API Manager, you can configure inbound security to invoke a policy defined in Policy Studio, see [Invoke Policy](https://docs.axway.com/bundle/axway-open-docs/page/docs/apim_administration/apimgr_admin/api_mgmt_virtualize_web/index.html). In cases like these the Discovery Agent will recognize the security being used and create an Access Request Definition that will display the authentication being used as well as the description of the policy being used.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connect_api_manager/gateway-administation.md:200]: Comma between "these" and "the Discovery Agent..."
Also,  you could end the first sentence after Access Request Defintion.  Then the next sentence could start with "The Access Request Defintion will display the authentication being used....

File: content/en/docs/connect_manage_environ/connect_api_manager/gateway-administation.md [hunk 3]
```diff
@@ -195,6 +195,14 @@ When using OAuth (External) inbound security, API Manager requires a custom OAut
...
+
+If the policy does not have a description, or the policy references an external link or markdown file the Discovery Agent will use the value defined by the `APIMANAGER_INVOKEPOLICYDEFAULTDESCRIPTION` environment variable.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connect_api_manager/gateway-administation.md:202]: Comma between file and "the Discovery Agent..."

**Review by sbolosan — APPROVED** [review]

File: content/en/docs/connect_manage_environ/connect_api_manager/agent-variables.md [hunk 4]
```diff
@@ -98,6 +98,7 @@ You can extend the previous minimum variable list with the following variables.
...
 | APIMANAGER_WORKERS                                                 | Controls how many Front End Proxies will be processed at a single time (default: 10).                                                                                                                                                                                                                                                                                                                                                                                                                      |
+| APIMANAGER_INVOKEPOLICYDEFAULTDESCRIPTION                          | When a proxy is secured by a policy, this description is added to the Access Request Definition if no desription text is found in API Manager (default: `Contact your provider about authenticating to this proxy`)                                                                                                                                                                                                                                                                                        |
```
Comment (vivekschauhan) [inline content/en/docs/connect_manage_environ/connect_api_manager/agent-variables.md:101]: "if no desription" -> "if no description"

**Review by vivekschauhan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by robertengaxway — APPROVED** [review]
I really like the references to actions which need to be taken on the V7 Manager to configure Invoke Policy to display correctly

**Review by sbolosan — APPROVED** [review]

**Comment by lbadenhop** [comment]
@jcollins-axway, please check my edits to line 208 in gateway-administration.md.

**Comment by jcollins-axway** [comment]
> @jcollins-axway, please check my edits to line 208 in gateway-administration.md.

looks good to me, thanks!

**Pull request merged**